sdk/core/azure-core-http-jdk11-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/implementation/JdkHttpClientProxySelector.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.jdk.httpclient.implementation;
+
+import java.io.IOException;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.SocketAddress;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * This class handles selecting the proxy during a request.
+ */
+public final class JdkHttpClientProxySelector extends ProxySelector {
+    private final Proxy.Type proxyType;
+    private final SocketAddress proxyAddress;
+    private final Pattern nonProxyHostsPattern;
+
+    public JdkHttpClientProxySelector(Proxy.Type proxyType, SocketAddress proxyAddress, String nonProxyHosts) {
```

Comment:
Constructor Javadoc?

Reply:
It's an implementation package type but added minimal Javadoc for this.